typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,628 @@
+
+        }
+        return null;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async registerFilesArtifacts(filesArtifactUrls: Map<FilesArtifactID, string>): Promise<Error> {
+        const filesArtifactIdStrsToUrls: Map<string, string> = new Map();
+        for (let [artifactId, url] of filesArtifactUrls.entries()) {
+            filesArtifactIdStrsToUrls[String(artifactId)] = url;
+        }
+        const args: RegisterFilesArtifactsArgs = newGetRegisterFilesArtifactsArgs(filesArtifactIdStrsToUrls);
+        
+        // TODO TODO TODO - CALLBACK & ERROR-HANDLING (REMOVE)
+        // if _, err := networkCtx.client.RegisterFilesArtifacts(context.Background(), args); err !== nil {
+        //     return stacktrace.Propagate(err, "An error occurred registering files artifacts: %+v", filesArtifactUrls)
+        // }
+
+        const promiseAsyncRegisterFilesArtifacts: Promise<ResultAsync<any, Error>> = new Promise((resolve, _unusedReject) => { //TODO
+            this.client.registerFilesArtifacts(args, (_unusedError: grpc.ServiceError, response: any) => {
+                resolve(okAsync(response));
+            })
+        });
+        const promiseRegisterFilesArtifacts: Result<any, Error> = await promiseAsyncRegisterFilesArtifacts;
+        if (!promiseRegisterFilesArtifacts.isOk()) {
+            return promiseRegisterFilesArtifacts.error;
+        }
+
+        return null;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async addService(
+        serviceId: ServiceID,
+        containerCreationConfig: ContainerCreationConfig,
+        generateRunConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => [ContainerRunConfig, Error]
+    ): Promise<[ServiceContext, Map<string, PortBinding>, Error]> {
+
+        const [serviceContext, hostPortBindings, err] = await this.addServiceToPartition(
+            serviceId,
+            defaultPartitionId,
+            containerCreationConfig,
+            generateRunConfigFunc,
+        );
+        if (err !== null) {
+            return [null, null, err];
+        }
+
+        return [serviceContext, hostPortBindings, null];
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async addServiceToPartition(
+        serviceId: ServiceID,
+        partitionId: PartitionID,
+        containerCreationConfig: ContainerCreationConfig,
+        generateRunConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => [ContainerRunConfig, Error],
+    ): Promise<[ServiceContext, Map<string, PortBinding>, Error]> {
+
+        winston.info("Registering new service ID with Kurtosis API..."); //TODO logrus.Trace is meant for something for low level, but I could find winston.info to be closest to this
```

Comment:
Omg that's so ridiculously annoying. Trace, debug, info, warn, error, crit are SO standardized across languages, but of _course_ the most popular logging framework in JS wants to do things completely differently.

The other thing that's annoying is, Winston _really_ wants you to create a logger object and pass it around everywhere (which we want to avoid if possible).

I dug through a fair few logging frameworks, and I think we actually want to switch to using loglevel ( https://github.com/pimterry/loglevel ). Sorry for the change :/ I think it will be much simpler for our use-case though